### PR TITLE
Make versions and source URLs of Kafka and Zookeeper configurable. Fix Java installation. 

### PR DESCRIPTION
**Pull request by milanjaros**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog
* Add version and source URL parameters.
* Make versions and source URLs of Kafka and Zookeeper configurable.
* Update the Ubuntu version.
* Error handling when it fails to download. Enhance help and make options optional.
* Make zookeeper version optional. 
* Move script. 
* Clean-up.

### CODE REVIEW

**Comment by milanjaros** [comment]
Hi @bmoore-msft, unfortunately, I am not able to locate and fix the "concat" issue. Could you point me to the exact place, where the function causes the problem, please?

File: kafka-on-ubuntu/azuredeploy.json [hunk 1]
```diff
@@ -185,14 +197,17 @@
           },
           "subnetPrefix": {
             "value": "[parameters('subnetPrefix')]"
+          },
+          "tshirtSizes": {
+            "value": "[concat('List of sizes because of test for Unreferenced variable in CI - it is not needed for proper function.', variables('tshirtSizeS'), variables('tshirtSizeM'), variables('tshirtSizeL'))]"
```
Comment (bmoore-msft) [inline kafka-on-ubuntu/azuredeploy.json]: This isn't going to work because your t-shirt size variables are json objects... they must be strings or arrays.

**Comment by bmoore-msft** [comment]
I added a comment on the line where I think the error is coming from...

**Comment by bmoore-msft** [comment]
Closing abandoned request - please reopen if you can address the feedback.

**Pull request closed without merging**